blog/2015/09/thinking-about-teaching.html

```diff
@@ -0,0 +1,360 @@
+  to find out what they're actually doing and how it has been going.
+  <a href="python-lesson-survey-results.pdf">The results</a> are fascinating&mdash;here
+  are some of the key findings, and my comments:
+</p>
+<dl>
+  <dt>
+    60% of our lessons are being taught to novices.
+  </dt>
+  <dd>
+    <p>
+      The term "novice" is a bit ambiguous here:
+      it can mean either "someone who has never programmed before"
+      or "someone who hasn't programmed <em>in Python</em>".
+      When we re-design this lesson,
+      we could well wind up creating different versions for these audiences,
+      since someone who's learning Python as a second language will be able to move faster and farther
+      than someone who is new to ideas like "variable", "list", and "function".
+    </p>
+  </dd>
+  <dt>
+    Roughly half of the respondents don't like the way we jump right
+    into an example.
+  </dt>
+  <dd>
+    <p>
+      The intention was to get to an authentic task as quickly as possible,
+      and "load CSV, create plot" is something every scientist does.
+      I still think it works for people who are learning Python as a second language,
+      but if that's only 1/3 of our workshops,
+      it's not surprising that it often doesn't work.
+    </p>
+  </dd>
+  <dt>
+    Only 1/3 of respondents got through the whole Python lesson.
+  </dt>
+  <dd>
+    <p>
+      Which means we need to cut material&mdash;even if we say,
+      "It's OK to go as far as you can,"
+      instructors will often feel that they've let their audience down
+      if they don't get through it all.
+    </p>
+  </dd>
+  <dt>
+    Over 80% of the respondents are teaching our materials with little
+    or no modification, but 22% are using an older version.
+  </dt>
+  <dd>
+    <p>
+      Based on a handful of conversations,
+      I think people use older material because
+      (a) they stick to whatever lesson they first taught,
+      and/or (b) they don't like the NumPy-early approach of the current lesson.
+      I don't know what we can do about (a),
+      but our "Python as a first language" lesson
+      should start with simpler tasks.
+    </p>
+  </dd>
+  <dt>
+    The defensive programming section which is skipped by over 85% of
+    the people who answered the survey, so we should move this
+    material elsewhere.
+  </dt>
+  <dd>
+    <p>
+      I (reluctantly) agree,
+      at least for people who are learning programming for the first time.
+      As with <a href="{{page.root}}/blog/2014/10/why-we-dont-teach-testing.html">testing</a>,
+      there's no point putting scarce classroom or maintenance time into something that doesn't stick.
+    </p>
+  </dd>
+  <dt>
+    82% of the people answering the survey are teaching out of the
+    Jupyter Notebook.
+  </dt>
+  <dd>
+    <p>
+      I was surprised by that, but I now wonder if it explains why it
```

Comment:
I'd love you to write more about this. I thought the notebook was preferred over scripting because it was more like a GUI interface which is closer to what students are used to. Scripts from the command line seem a little more magical and losing the interactivity seems like a step back. Maybe this should be a separate blog post, but I'd love to hear more about it